nixos/modules/services/web-servers/tomcat.nix

```diff
@@ -185,6 +188,11 @@ in
       after = [ "network.target" ];
 
       preStart = ''
+        # Delete everything but logs from the existing base directory
+        # (we don't want remainders of old configuration!)
+        [[-e ${cfg.baseDir} ]] \
+          && find ${cfg.baseDir} -mindepth 1 -maxdepth 1 -not -name logs -exec rm -rf '{}' '+'
```

Comment:
Is it possible to do it more explicit?

```
rm -rf ${cfg.baseDir}/{conf,virtualhosts,temp,lib,shared/lib,webapps,work}
```

Are there any files in those directory that should be not deleted? I am not an expert when it comes to tomcat related applications.

Reply:
> Is it possible to do it more explicit?

Yes, this can work as well. Though I would also include `bin` which is a symlink created by the script. And `work` needs to be excluded (see below).

> Are there any files in those directory that should be not deleted?

In fact, there are. The session files, named `SESSIONS.ser` by default, can exist inside the `work` directory, and they shouldn't be deleted.

On the other hand, `work` serves as a cache for compiled JSP files, and in certain corner cases Tomcat can improperly use the cached files from an old deployment. But these corner cases should be very uncommon (if at all possible), so probably it doesn't make much spending code lines on handling them.